detection_rules/rule_validators.py

```diff
@@ -37,28 +42,39 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
             return
 
         indexes = data.index or []
-        beats_version = meta.beats_version or beats.get_max_version()
-        ecs_versions = meta.ecs_versions or [ecs.get_max_version()]
+        stack_versions = get_stack_schemas(meta.min_stack_version)
+
+        if not stack_versions:
+            # add latest
+            stack_versions = {CURRENT_PACKAGE: {'beats': beats.get_max_version(), 'ecs': ecs.get_max_version()}}
+
+        if meta.query_validate_master:
+            stack_versions['latest'] = {'beats': 'master', 'ecs': 'master'}
 
         beat_types = [index.split("-")[0] for index in indexes if "beat-*" in index]
-        beat_schema = beats.get_schema_from_kql(ast, beat_types, version=beats_version) if beat_types else None
 
-        if not ecs_versions:
-            kql.parse(self.query, schema=ecs.get_kql_schema(indexes=indexes, beat_schema=beat_schema))
```

Comment:
It looks like this versionless validation is gone now?
What are the implications?

Reply:
version-less default to the highest non-master locally saved, so this is accounted for